### PR TITLE
Fix deps conversion

### PR DESCRIPTION
**Pull request by alexandresgf**
Since Pipenv v2022.8.30 the function `convert_deps_to_pip` returns a list of strings containing all the dependencies found in the Pipfile converted to Pip format. So, I added a try/exception statement to catch the `TypeError` exception, when the previous code tries to open the temporary file, and write the dependencies to the requirements file.

### CODE REVIEW

**Comment by elacuesta** [comment]
Hi, thanks for the fix. I have a few comments:
* Please split the code paths based on whether the received argument is a list - I don't think catching `TypeError` makes this distinction clear.
* Joining a variable called `_reqs_file` is confusing to me, please rename the parameter to show that it can be either a string or a list. It would be even better if you could also add type hints.
* Could this be tested? I guess it would be enough to add an environment with a different pipenv version, without actually modifying any test cases.

**Comment by alexandresgf** [comment]
> * Could this be tested? I guess it would be enough to add an environment with a different pipenv version, without actually modifying any test cases.

I can mock-up the behavior of both `convert_deps_to_pip` and write the tests. Is it sounds good?

**Comment by elacuesta** [comment]
Sounds good to me, thanks!

File: shub/deploy.py [hunk 1]
```diff
@@ -169,7 +170,8 @@ def _get_pipfile_requirements(tmpdir=None):
         from pipenv.utils.indexes import prepare_pip_source_args
     except ImportError:
         try:
-            from pipenv.utils import convert_deps_to_pip, prepare_pip_source_args
+            from pipenv.utils import (convert_deps_to_pip,
+                                      prepare_pip_source_args)
```
Comment (elacuesta) [inline shub/deploy.py:174]: Please revert this change, we don't have such line length restriction in place.

File: shub/deploy.py [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
 # Not used in code but needed in runtime, don't remove!
 import setuptools
 import setuptools.msvc  # noqa
+import toml
+from six.moves.urllib.parse import urljoin
 
-from shub.config import (list_targets_callback, load_shub_config,
-                         SH_IMAGES_REGISTRY)
+from shub.config import (SH_IMAGES_REGISTRY, list_targets_callback,
+                         load_shub_config)
```
Comment (elacuesta) [inline shub/deploy.py:18]: Please put this in one line

File: shub/deploy.py [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
-import json
-from six.moves.urllib.parse import urljoin
+from typing import AnyStr, Optional, Union
 
 import click
-import toml
 # Not used in code but needed in runtime, don't remove!
 import setuptools
 import setuptools.msvc  # noqa
+import toml
```
Comment (elacuesta) [inline shub/deploy.py:14]: I appreciate the attempt to clean up the imports, however let's keep the whiteline after the setuptools imports - the comment above only applies to them.

**Review by elacuesta — COMMENTED** [review]
Great job, thanks. I left a minor comments.

File: shub/deploy.py (hunk 3 above)
Comment (alexandresgf) [reply shub/deploy.py:14]: Sorry, it is an addiction that I have to format the code after making any changes. I have already returned it to the previous state.

File: shub/deploy.py (hunk 2 above)
Comment (alexandresgf) [reply shub/deploy.py:18]: Sorry, it is an addiction that I have to format the code after making any changes. I have already returned it to the previous state.

File: shub/deploy.py (hunk 1 above)
Comment (alexandresgf) [reply shub/deploy.py:174]: Sorry, it is an addiction that I have to format the code after making any changes. I have already returned it to the previous state.

**Comment by elacuesta** [comment]
Thanks @alexandresgf :partying_face:
